### DISCUSSION TITLE
Rancher Desktop Design - Settings

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#691] [Archive]
## Settings

* Rename "Kubernetes Settings" to "Settings"
* Introduce two sections 
   * Kubernetes Settings
      > For managing things like Kubernetes Version and Port 
   * Resources
      > For modifying memory and # of cpu cores
* Replace "Reset Kubernetes" button with Apply & Cancel
* Move "Reset Kubernetes" button to Troubleshooting - Under Review
   > Some discussions have revealed that this is not desirable, as having quick access for resetting Kubernetes and container images is beneficial for a lot of use cases. Hiding this feature behind troubleshooting might not be the best option here
* Move "Supporting Utilities" to its own menu item

![3 - Settings](https://user-images.githubusercontent.com/835961/134972929-79f03049-c7d9-4bcf-90dd-fa705313eaec.png)

> Using the space to share this initial design proposal from the UI group. I've add some quick notes for each section of Rancher Desktop that might change. If you have any questions about anything you see below, please ask and I'll provide some more detail. 
>
> Original Discussion #683
>
> Feedback is highly valued and welcome.

**Comment by jandubois** [comment]
> ## Settings
> * Rename "Kubernetes Settings" to "Settings"
> * Introduce two sections
>   
>   * Kubernetes Settings
>     > For managing things like Kubernetes Version and Port
>   * Resources
>     > For modifying memory and # of cpu cores

This is a bit hard to discuss, as we haven't reached consensus yet how we want to support multiple container runtimes etc. So the following is just my vision on how I think it should work; I believe @mattfarina disagrees with at least some of my assumptions.

We are going to have multiple "runtimes" (for the lack of a better term right now), like

* Kubernetes
* Docker
* Containerd
* Podman

Each one will have their own settings that are independent from the other runtimes, so instead of renaming "Kubernetes Settings" to "Settings" I think it rather should be renamed to "Kubernetes".

I think we should have a global "Settings" page (maybe different name though), just below "Welcome". It would have a checkbox for each of the runtimes supported. As you check any of the boxes, the UI will add a settings page for each of the runtimes: "Kubernetes", "Docker", "Containerd", and "Podman".

I'm struggling to find a fitting name for "Resources": I think it is more general, as it should contain all the settings that are global to the virtual machine, and therefore shared by all the runtimes. So yes, memory and cpu settings, but in the future also maybe volume mounts, network configuration (dns and proxy settings), certificates, etc.

I want to avoid calling it "VM Settings", but maybe "Global", "Common", "Shared" settings? Either way, this page should be above the individual runtime settings, possibly directly behind "Welcome".

Comment copied from https://github.com/rancher-sandbox/rancher-desktop/discussions/683#discussioncomment-1393134

**Comment by jandubois** [comment]
I'm going to skip discussing the individual runtime settings pages in detail here, but just as an example, settings for "Kubernetes" would include things like:

* Use containerd or dockerd runtime
* enable traefik
* enable metrics
* (everything exposed by `k3s` that is meaningful to customize and doesn't break RD)

Comment copied from https://github.com/rancher-sandbox/rancher-desktop/discussions/683#discussioncomment-1393139

**Comment by mook-as** [comment]
There a single Apply button for the whole page; does that mean none of the changes take place until the user clicks on that button?  Should that button container somehow reference both sections, and not just the Resources section?

(I'm not convinced that trying to support _all_ runtimes at once is a practical idea, but that's a separate discussion…)

**Reply by rak-phillip** [reply]
The intent is that changes here wouldn't take place until `Apply` is clicked, which falls in line with the current behavior of this area where `Reset Kubernetes`/`Reset Kubernetes and Container Images` is clicked (or accepting the prompt when changing Kubernetes version) before changes take effect. 

I'll see what we can come up with to make it a little more explicit that settings changed in any given section will be applied.
